File: Task2/apz-pzpi-21-11-milennyi-ivan-task2/src/models/sheep.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use utoipa::ToSchema;
use validator::Validate;

#[derive(Debug, Validate, Deserialize, Serialize, FromRow, ToSchema)]
pub(crate) struct Sheep{
    id: Option<u64>,
    birth_date: u64,
    breed_id: u64,
    weight: u64,
    sex: bool,
    temperature_scanner_id: Option<u64>,
    shepherd_id: Option<u64>
}
impl Sheep {
    pub fn id(&self) -> Option<u64> {
        self.id
    }
    pub fn set_id(&mut self, id: u64) -> (){
        self.id = Some(id);
    }
    pub fn birth_date(&self) -> u64 {
        self.birth_date
    }

    pub fn breed_id(&self) -> u64 {
        self.breed_id
    }
    pub fn weight(&self) -> u64{
        self.weight
    }
    pub fn sex(&self) -> bool {
        self.sex
    }

    pub fn temperature_scanner_id(&self) -> Option<u64> {
        self.temperature_scanner_id
    }

    pub fn shepherd_id(&self) -> Option<u64> {
        self.shepherd_id
    }
}

File: Task4/apz-pzpi-21-11-milennyi-ivan-task4/src/models/Sheep.ts
export class Sheep {
    id: number | null;
    birth_date: number;
    breed_id: number;
    weight: number;
    sex: boolean;
    temperature_scanner_id: number | null;
    shepherd_id: number | null;

    constructor(id: number | null, birthDate: number, breedId: number, weight: number, sex: boolean, temperatureScannerId: number | null, shepherdId: number | null) {
        this.id = id;
        this.birth_date = birthDate;
        this.breed_id = breedId;
        this.weight = weight;
        this.sex = sex;
        this.temperature_scanner_id = temperatureScannerId;
        this.shepherd_id = shepherdId;
    }
}